src/Microsoft.AspNet.Http.Abstractions/HttpResponse.cs

```diff
@@ -24,9 +25,14 @@ public abstract class HttpResponse
 
         public abstract bool HasStarted { get; }
 
-        public abstract void OnResponseStarting(Action<object> callback, object state);
+        public abstract void OnResponseStarting(Func<object, Task> callback, object state);
 
-        public abstract void OnResponseCompleted(Action<object> callback, object state);
+        public void OnResponseStarting(Func<object, Task> callback)
+        {
+            OnResponseStarting(callback, state: null);
+        }
+
```

Comment:
How about `OnResponseStarting(Func<Task> callback) `?

Sure, add the same for OnResponseCompleted too.